Add Username Prefix Enforce for DynamicFile mode (#568)
kubernetes-sigs/aws-iam-authenticator: README.md

running EKS in addition to some other AWS cluster(s) and want to have the same
mappings in each.

#### `DynamicFile`

> A local file specified by cfg.dynamicfilepath can serve as backend. The file

Reviewer: ```suggestion
A local file specified by cfg.dynamicfilepath can serve as the backend. The file
```

> content is expected to be in exactly the same format as in EKSConfigMap. Whenever

Reviewer: ```suggestion
content is expected to be in exactly the same format as the EKSConfigMap. Whenever
```

> This file content changes, authenticator will automatically reload it.  This 

Reviewer: ```suggestion
this file content changes, authenticator will automatically reload it.  This 
```

> provides more flexibility on managing the arn mappings.

Reviewer: ```suggestion
provides more flexibility on managing the ARN mappings.
```

check https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/hack/dev/authenticator_with_dynamicfile_mode.yaml 
about how to configure the DynamicFile mode.

run `make e2e RUNNER=kind` to play with a kind cluster with DynamicFile mode enable.

> ### 5. how to configure reservedPrefixConfig for k8s username

Reviewer: ```suggestion
### 5. How to configure reservedPrefixConfig for Kubernetes usernames
```

> check https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/hack/dev/authenticator_with_dynamicfile_mode.yaml 
> about how to configure the DynamicFile mode.
> 
> run `make e2e RUNNER=kind` to play with a kind cluster with DynamicFile mode enable.
> 
> ### 5. How to configure reservedPrefixConfig for Kubernetes usernames
> 
> authenticator can support reserved prefix for k8s username. if the reserved prefix is
> set, then username with the reserved prefix will not be authenticated with the error
> "username must not begin with with the following prefixes:".
> 
> check https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/hack/dev/authenticator_with_dynamicfile_mode.yaml
> about how to configure the reserved prefix. 

Reviewer 2: ```suggestion
Check https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/hack/dev/authenticator_with_dynamicfile_mode.yaml 
about how to configure the DynamicFile mode.

Run `make e2e RUNNER=kind` to play with a kind cluster with DynamicFile mode enable.

### 5. How to configure reservedPrefixConfig for Kubernetes usernames

The aws-iam-authenticator can support reserved prefix for k8s username. If the reserved prefix is
set, then the username with the reserved prefix will not be authenticated with the error
"username must not begin with with the following prefixes:".

Check https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/master/hack/dev/authenticator_with_dynamicfile_mode.yaml
about how to configure the reserved prefix. 
```
